### ISSUE TITLE
dialyzer: unknown functions and unknown types warnings are not suppressed

### ISSUE THREAD

**Issue by ilya-klyuchnikov** [erlang/otp#8822]
**Describe the bug**

Dialyzer CLI:
```
  -Wno_unknown
     Suppress warnings about unknown functions and types. The default is to
     warn about unknown functions and types when setting the exit
     status. When using Dialyzer from Erlang, warnings about unknown functions
     and types are returned.
```

Running dialyzer for the following module:

```erlang
-module(dial).
-export([foo/1]).

-dialyzer({nowarn_function, foo/1}).
-spec foo(bar:bar()) -> bar:bar().
foo(T) ->
  bar:bar(T).
```

```
dialyzer -Wno_unknown dial.erl
Unknown functions:
  bar:bar/1 (dial.erl:7:3)
Unknown types:
  bar:bar/0 (dial.erl:5:11)
 done in 0m0.12s
done (passed successfully)
```

Having `{warnings, [no_unknown]}.` in `dialyzer.config` also doesn't help.
So, warnings about unknown types/functions are not suppressable.

**Expected behavior**
Such warnings are suppressed when corresponding suppressions are turned on.

**Affected versions**
OTP 26 and 27.

**Comment by essen** [comment]
I have seen this as well. OTP-27 added/improved some warnings, I wanted to silence them as a temporary measure, and could not. Instead I tricked Dialyzer by doing `Mod = bar, Mod:bar(T)`.

**Comment by jhogberg** [comment]
Thanks for your report, I've fixed it in #8885.
